crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -159,21 +159,18 @@ impl AuthorityPerpetualTables {
         &self,
         object_id: ObjectID,
         version: SequenceNumber,
-    ) -> Option<Object> {
-        let Ok(iter) = self
+    ) -> SuiResult<Option<Object>> {
+        let iter = self
             .objects
             .safe_range_iter(ObjectKey::min_for_id(&object_id)..=ObjectKey::max_for_id(&object_id))
-            .skip_prior_to(&ObjectKey(object_id, version))
-        else {
-            return None;
-        };
-        iter.reverse().next().and_then(|db_result| match db_result {
+            .skip_prior_to(&ObjectKey(object_id, version))?;
+        Ok(iter.reverse().next().and_then(|db_result| match db_result {
             Ok((key, o)) => self.object(&key, o).ok().flatten(),
             Err(err) => {
```

Comment:
Should we return this error instead of ignoring it then?

Reply:
Oh wait I didn't even notice that one! Thanks!